Libs/RaptureAddins/ElasticSearch/src/test/java/rapture/elasticsearch/ElasticSearchSearchRepositoryTest.java

```diff
@@ -250,19 +261,138 @@ public void testRemove() {
+        // ImmutableList.of("16", "Crystal Palace", "36", "-10", "39"),
+        // ImmutableList.of("17", "Newcastle", "36", "-25", "33"),
+        // ImmutableList.of("18", "Sunderland", "35", "-18", "32"),
+        // ImmutableList.of("19", "Norwich", "35", "-26", "31"),
+        // ImmutableList.of("20", "Aston Villa", "36", "-45", "16"));
+        // String leagueJson = JacksonUtil.jsonFromObject(leagueList);
+
+        // Unfortunately it seems that ElasticSearch requires a JSON document.
+        // It won't accept a JSON list or a CSV. So we have to munge it.
+
+        // This is a minimal PDF. Next step is to have one with data in it and show that we can index the contents
+        String minimalPdf = "%PDF-1.0\n"
+                + "1 0 obj<</Type/Catalog/Pages 2 0 R>>endobj 2 0 obj<</Type/Pages/Kids[3 0 R]/Count 1>>endobj 3 0 obj<</Type/Page/MediaBox[0 0 3 3]>>endobj\n"
+                + "xref\n" + "0 4\n" + "0000000000 65535 f\n" + "0000000010 00000 n\n" + "0000000053 00000 n\n" + "0000000102 00000 n\n"
+                + "trailer<</Size 4/Root 1 0 R>>\n" + "startxref\n" + "149\n" + "%EOF\n";
+
+        Map<String, String> copout = new HashMap<>();
+        copout.put("premier", premier);
+        String leagueJson = JacksonUtil.jsonFromObject(copout);
+        RaptureURI epl = new RaptureURI.Builder(Scheme.BLOB, "unittest").docPath("English/Premier").build();
+        e.put(new BlobUpdateObject(epl, leagueJson.getBytes(), MediaType.JSON_UTF_8.toString()));
+        // e.put(new BlobUpdateObject(epl, premier.getBytes(), MediaType.JSON_UTF_8.toString()));
+
+        RaptureURI champ = new RaptureURI.Builder(Scheme.BLOB, "unittest").docPath("English/Championship").build();
+        e.put(new BlobUpdateObject(champ, championship.getBytes(), MediaType.CSV_UTF_8.toString()));
+
+
+        // Try reading a real PDF from a file
+
+        File pdf = new File("src/test/resources/www-bbc-com.pdf");
+        try {
+            byte[] content = Files.readAllBytes(pdf.toPath());
+            RaptureURI firstDiv = new RaptureURI.Builder(Scheme.BLOB, "unittest").docPath("English/First").build();
+            e.put(new BlobUpdateObject(firstDiv, content, MediaType.PDF.toString()));
+        } catch (IOException e2) {
+            fail(pdf.getAbsolutePath() + " : " + ExceptionToString.format(e2));
+        }
+
+        File pdf2 = new File("src/test/resources/HelloWorld.pdf");
+        try {
+            byte[] content = Files.readAllBytes(pdf2.toPath());
+            RaptureURI hello = new RaptureURI.Builder(Scheme.BLOB, "Hello").docPath("Hello/Wurld").build();
+            e.put(new BlobUpdateObject(hello, content, MediaType.PDF.toString()));
+        } catch (IOException e3) {
+            fail(pdf2.getAbsolutePath() + " : " + ExceptionToString.format(e3));
+        }
 
+        try {
+            Thread.sleep(3000);
+            // Indexing happens in a separate thread so give it a chance
```

Comment:
call e.refresh() in your unit tests after any async operation, so you dont have to wait